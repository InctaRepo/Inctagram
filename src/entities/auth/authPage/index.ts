export { AuthPage } from 'src/entities/auth/authPage/ui/AuthPage'
