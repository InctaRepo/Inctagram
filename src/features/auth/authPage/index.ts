export { AuthPage } from './ui/AuthPage'
