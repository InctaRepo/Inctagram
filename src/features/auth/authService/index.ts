export type { UserType, AccessType, AuthState } from './authApiTypes'
export { authReducer, setLogout, setIsAuth, setUser } from './authSlice'
