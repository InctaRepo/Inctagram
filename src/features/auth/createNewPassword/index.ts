export { CreateNewPassword } from './ui/CreateNewPassword'
