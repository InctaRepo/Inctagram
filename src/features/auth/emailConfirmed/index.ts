export { EmailConfirmed } from './ui/EmailConfirmed'
