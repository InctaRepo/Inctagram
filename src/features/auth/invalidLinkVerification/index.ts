export { InvalidLinkVerification } from './ui/InvalidLinkVerification'
