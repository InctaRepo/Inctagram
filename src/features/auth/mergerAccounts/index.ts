export { MergerAccounts } from './ui/MergerAccounts'
