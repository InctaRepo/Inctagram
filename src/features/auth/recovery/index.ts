export { Recovery } from '@/features/auth/recovery/ui/Recovery'
