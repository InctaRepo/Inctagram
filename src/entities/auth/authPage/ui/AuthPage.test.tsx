import React from 'react'

import { customRender as render } from '@/__mocks__/customRender'
import { AuthPage } from '@/entities/auth/authPage'
import { screen, waitFor } from '@testing-library/react'
import { userEvent } from '@testing-library/user-event'
import clsx from 'clsx'

const routerMock = { locale: 'en', push: '/dashboard' }

jest.mock('next/router', () => ({
  useRouter: jest.fn(() => routerMock),
}))

function setup(jsx: React.JSX.Element) {
  return {
    user: userEvent.setup(),
    ...render(jsx),
  }
}

describe('AuthPage', () => {
  beforeEach(() => {
    jest.resetAllMocks()
  })

  it('should render the component with the correct props', async () => {
    const linkPathMock = '/dashboard'
    const nameButtonMock = 'Login'
    const nameButtonTwoMock = 'Register'
    const textMock = 'Welcome to the authentication page!'
    const titleMock = 'Authentication'
    const variantMock = 'merger'

    render(
      <AuthPage
        linkPath={linkPathMock}
        nameButton={nameButtonMock}
        nameButtonTwo={nameButtonTwoMock}
        text={textMock}
        title={titleMock}
        variant={variantMock}
      />
    )
    expect(screen.getByText(titleMock)).toBeInTheDocument()
    expect(screen.getByText(textMock)).toBeInTheDocument()
    expect(screen.getByText(nameButtonMock)).toBeInTheDocument()
    expect(screen.getByText(nameButtonTwoMock)).toBeInTheDocument()

    const buttonOne = screen.getByText(nameButtonMock).parentElement
    const buttonTwo = screen.getByText(nameButtonTwoMock).parentElement

    expect(buttonOne).toHaveClass(clsx('button', 'buttonMerger'))
    expect(buttonTwo).toHaveClass('button')
  })

  it('should render the component without the second button when variant prop is not provided', async () => {
    const linkPathMock = '/dashboard'
    const nameButtonMock = 'Login'
    const nameButtonTwoMock = 'Register'
    const textMock = 'Welcome to the authentication page!'
    const titleMock = 'Authentication'
    const variantMock = undefined
    const { debug, user } = setup(
      <AuthPage
        linkPath={linkPathMock}
        nameButton={nameButtonMock}
        nameButtonTwo={nameButtonTwoMock}
        text={textMock}
        title={titleMock}
        variant={variantMock}
      />
    )

    expect(screen.getByText(titleMock)).toBeInTheDocument()
    expect(screen.getByText(textMock)).toBeInTheDocument()
    expect(screen.getByText(nameButtonMock)).toBeInTheDocument()
    expect(screen.queryByText(nameButtonTwoMock)).toBeNull()

    const buttonOne = screen.getByText(nameButtonMock).parentElement

    expect(buttonOne).toHaveClass('button')

    user.click(screen.getByRole('button', { name: 'Login' }))
    waitFor(() => expect(routerMock.push).toHaveBeenCalledWith(linkPathMock))
  })
})
