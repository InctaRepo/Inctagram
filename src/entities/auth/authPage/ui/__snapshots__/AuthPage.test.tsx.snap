// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthPage snapshot AuthPage with merger 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div
          class="text_container"
        >
          <div
            class="title"
          >
            <p
              class="bold16 primary  undefined"
            >
              Authentication
            </p>
          </div>
          <div
            class="text"
          >
            <p
              class="regular14 primary  undefined"
            >
              Welcome to the authentication page!
            </p>
          </div>
          <button
            class="outlined fullWidth button buttonMerger"
          >
            <p
              class="bold14 primary  undefined"
            >
              Login
            </p>
          </button>
          <button
            class="outlined  button buttonMerger"
          >
            <p
              class="bold14 primary  undefined"
            >
              Register
            </p>
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div
        class="text_container"
      >
        <div
          class="title"
        >
          <p
            class="bold16 primary  undefined"
          >
            Authentication
          </p>
        </div>
        <div
          class="text"
        >
          <p
            class="regular14 primary  undefined"
          >
            Welcome to the authentication page!
          </p>
        </div>
        <button
          class="outlined fullWidth button buttonMerger"
        >
          <p
            class="bold14 primary  undefined"
          >
            Login
          </p>
        </button>
        <button
          class="outlined  button buttonMerger"
        >
          <p
            class="bold14 primary  undefined"
          >
            Register
          </p>
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
  "user": {
    "clear": [Function],
    "click": [Function],
    "copy": [Function],
    "cut": [Function],
    "dblClick": [Function],
    "deselectOptions": [Function],
    "hover": [Function],
    "keyboard": [Function],
    "paste": [Function],
    "pointer": [Function],
    "selectOptions": [Function],
    "setup": [Function],
    "tab": [Function],
    "tripleClick": [Function],
    "type": [Function],
    "unhover": [Function],
    "upload": [Function],
  },
}
`;

exports[`AuthPage snapshot AuthPage with undefined 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div
          class="text_container"
        >
          <div
            class="title"
          >
            <p
              class="bold16 primary  undefined"
            >
              Authentication
            </p>
          </div>
          <div
            class="text"
          >
            <p
              class="regular14 primary  undefined"
            >
              Welcome to the authentication page!
            </p>
          </div>
          <button
            class="primary fullWidth button"
          >
            <p
              class="bold14 primary  undefined"
            >
              Login
            </p>
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div
        class="text_container"
      >
        <div
          class="title"
        >
          <p
            class="bold16 primary  undefined"
          >
            Authentication
          </p>
        </div>
        <div
          class="text"
        >
          <p
            class="regular14 primary  undefined"
          >
            Welcome to the authentication page!
          </p>
        </div>
        <button
          class="primary fullWidth button"
        >
          <p
            class="bold14 primary  undefined"
          >
            Login
          </p>
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
  "user": {
    "clear": [Function],
    "click": [Function],
    "copy": [Function],
    "cut": [Function],
    "dblClick": [Function],
    "deselectOptions": [Function],
    "hover": [Function],
    "keyboard": [Function],
    "paste": [Function],
    "pointer": [Function],
    "selectOptions": [Function],
    "setup": [Function],
    "tab": [Function],
    "tripleClick": [Function],
    "type": [Function],
    "unhover": [Function],
    "upload": [Function],
  },
}
`;
